pixeltable/utils/hf_datasets.py

```diff
@@ -0,0 +1,111 @@
+                                            primary_key: Union[str, List[str]],
+                                            num_retained_versions: int) -> Optional['catalog.InsertableTable']:
+    """ See pixeltable.Client.create_table_from_huggingface_dataset for documentation
+    """
+
+    dataset_dict : Dict[str, datasets.Dataset] = {}
+    if isinstance(dataset, datasets.Dataset):
+        # get split string while removing any slice notation
+        if dataset.split._name is not None:
+            # when loading an hf dataset partially, dataset.split._name is sometimes the form "train[0:1000]"
+            # in those cases, we only only want to keep the "train" part
+            split_name = dataset.split._name.split('[')[0]
+        else:
+            split_name = None
+
+        if split_name is None and split_column_name is not None:
+            print('no split name found in dataset, ignoring')
+            split_column_name = None
+
+        dataset_dict = {split_name:dataset}
+    elif isinstance(dataset, datasets.DatasetDict):
+        for column_name in dataset:
+            dataset_dict[column_name] = dataset[column_name]
+    else:
+        raise excs.Error(f'type(dataset) must be datasets.Dataset or datasets.DatasetDict. Got {type(dataset)}')
+
+    # get the schema from the first split
+    dataset_features = next(iter(dataset_dict.values())).features
+
+    pixeltable_schema : Dict[str, ts.ColumnType] = {}
+    categorical_features : Dict[str, datasets.ClassLabel] = {}
+    for column_name,feature_type in dataset_features.items():
+        if isinstance(feature_type, datasets.ClassLabel):
+            # enum, example: ClassLabel(names=['neg', 'pos'], id=None)"
+            pixeltable_schema[column_name] = ts.StringType(nullable=True)
+            categorical_features[column_name] = feature_type.names
+        elif isinstance(feature_type, datasets.Value):
+            # example: Value(dtype='int64', id=None)
+            if feature_type.dtype in _hf_to_pt:
+                dest_type = _hf_to_pt[feature_type.dtype]
+                pixeltable_schema[column_name] = dest_type
+            else:
+                raise excs.Error(f'unsupported value type {feature_type} for column {column_name}')
+        else:
+            raise excs.Error(f'unsupported feature type {feature_type} for column {column_name}')
+
+
+    if split_column_name is not None:
+        if split_column_name in pixeltable_schema:
+            raise excs.Error(f'split column name {split_column_name} already exists in dataset schema, use a different name')
+        pixeltable_schema[split_column_name] = ts.StringType(nullable=False)
+
+    print('mapping huggingface features to pixeltable schema:', pixeltable_schema)
+    if explain_only:
+        return None
+
+    tab = self.create_table(path_str, pixeltable_schema, primary_key, num_retained_versions)
+
+    try:
+        target_bytes_per_batch = 20_000_000 # 20MB
```

Comment:
Where does this number come from? Should it be configurable?

Reply:
We just want to make sure we don't attempt to load the full dataset if it is large, and at the same time, avoid many calls with tiny data sizes (potential for large overhead). The exact number does not matter, I think being in the 10s of MB range is a good size for a lot of storage operations. I'd abstain from adding an external tunable unless a user would benefit from it.

Reply:
Ok. Maybe add a code comment explaining the justification behind this number, and factor it out as a constant in some more visible/global place in the code.